packages/assets-controllers/src/assetsUtil.ts

```diff
@@ -292,3 +296,49 @@ export async function reduceInBatchesSerially<
   const finalResult = workingResult as Result;
   return finalResult;
 }
+
+/**
+ * fetches and maps the exchange rates
+ * @param nativeCurrency - native currency
+ * @param tokenAddresses - array of token addresses
+ * @param chainId - The ID of the chain.
+ */
+export async function fetchAndMapExchangeRates(
```

Comment:
I see that the name for this function was taken from TokenRatesController, but I'm wondering if there's a better one we can use. The "map" part in particular seems a bit unnecessary. Also, it's not clear what kind of exchange rates are being fetched, so it seems important to say that they're for tokens.

Therefore, perhaps we call this `fetchTokenExchangeRates` or even `fetchTokenContractExchangeRates` (to match the fact that the type is called `ContractExchangeRates`)?

Reply:
thanks naming updated to fetchTokenContractExchangeRates !